src/hooks/useEntityPreviews.tsx

```diff
@@ -62,5 +65,17 @@ export function useEntityPreviews(
     entityPreviewSearcher.setUniversalLimit(universalLimit);
     debouncedUniversalSearch();
   }
-  return [{ verticalResultsArray, isLoading }, executeEntityPreviewsQuery];
+  return [{ verticalKeyToResults, isLoading }, executeEntityPreviewsQuery];
+}
+
+/**
+ * @returns a mapping of vertical key to VerticalResults
+ */
+function getVerticalKeyToResults(
+  verticalResultsArray: VerticalResultsData[]
+): Record<string, VerticalResultsData> {
+  return verticalResultsArray.reduce<Record<string, VerticalResultsData>>((prev, current) => {
+    prev[current.verticalKey] = current;
```

Comment:
do we want to just store `current.results` so it's just the `Result[]` instead of the whole `VerticalResultsData`?